packages/web-react/src/components/Collapse/Collapse.tsx

```diff
@@ -1,92 +1,47 @@
-import React, { createElement } from 'react';
+import React, { useRef, MutableRefObject } from 'react';
 import classNames from 'classnames';
 import { SpiritCollapseProps } from '../../types';
 import { useStyleProps } from '../../hooks/styleProps';
-import { useCollapse } from './useCollapse';
 import { useCollapseStyleProps } from './useCollapseStyleProps';
 import { useCollapseAriaProps } from './useCollapseAriaProps';
+import { useResizeHeight } from './useResizeHeight';
 
-const Collapse = (props: SpiritCollapseProps) => {
-  const {
-    id = Math.random().toString(36).slice(2, 7),
-    elementType = 'div',
-    contentElementType = 'div',
-    collapsibleToBreakpoint = undefined,
-    children,
-    isCollapsed,
-    hideOnCollapse,
-    renderTrigger,
-    ...rest
-  } = props;
-
-  const { collapsed, toggleHandler } = useCollapse({
-    isCollapsed,
-  });
-  const { wrapperClassName, contentClassName, triggerClassName } = useCollapseStyleProps({
-    isCollapsed: collapsed,
-  });
-  const { wrapperProps, contentProps, triggerProps } = useCollapseAriaProps({
-    id,
-    collapsibleToBreakpoint,
-    toggleHandler,
-    isCollapsed: collapsed,
-  });
-
-  const { styleProps: wrapperStyleProps, props: wrapperOtherProps } = useStyleProps({ ...rest });
-  const { styleProps: triggerStyleProps } = useStyleProps({
-    UNSAFE_className: triggerClassName,
-  });
-
-  const content = createElement(
-    contentElementType,
-    {
-      ...contentProps,
-      className: contentClassName,
-    },
-    children,
-  );
-
-  const triggerRenderHandler = () => {
-    const show = hideOnCollapse ? !(hideOnCollapse && collapsed) : true;
-    if (renderTrigger && show)
-      return renderTrigger({
-        collapsed,
-        trigger: {
-          ...triggerStyleProps,
-          ...triggerProps,
-        },
-      });
+const defaultProps = {
+  isCollapsed: false,
+  collapsibleToBreakpoint: undefined,
+};
 
-    return null;
-  };
+const Collapse = (props: SpiritCollapseProps) => {
+  const { children, ...restProps } = props;
 
-  const wrapperRenderHandler = () => {
-    if (hideOnCollapse && collapsed) {
-      return children;
-    }
+  const collapseElementRef: MutableRefObject<HTMLElement | undefined> = useRef();
```

Comment:
Try `HTMLDivElement` instead of `HTMLElement` because the element you set the ref to is a `div`

Reply:
In this case you are right. I leave it because I did not know if we will stay with changing this inner element.